lib/ui/painting.dart

```diff
@@ -4935,26 +4591,1632 @@ class PictureRecorder extends NativeFieldWrapperClass2 {
+      if (_curStrokeWidth != paint.strokeWidth) {
+        _curStrokeWidth = paint.strokeWidth;
+        _addOp(_CanvasOp.setStrokeWidth, 1);
+        _addScalar(_curStrokeWidth);
+      }
+      if (_curStrokeCap != paint.strokeCap) {
+        _curStrokeCap = paint.strokeCap;
+        _addOp(_strokeCapOps[_curStrokeCap.index], 0);
+      }
+      if (_curStrokeJoin != paint.strokeJoin) {
+        _curStrokeJoin = paint.strokeJoin;
+        _addOp(_strokeJoinOps[_curStrokeJoin.index], 0);
+      }
+      if (_curMiterLimit != paint.strokeMiterLimit) {
+        _curMiterLimit = paint.strokeMiterLimit;
+        _addOp(_CanvasOp.setMiterLimit, 1);
+        _addScalar(_curMiterLimit);
+      }
+    }
+    if ((dataNeeded & _filterQualityNeeded) != 0 && _curFilterQuality != paint.filterQuality) {
+      _curFilterQuality = paint.filterQuality;
+      _addOp(_filterQualityOps[_curFilterQuality.index], 0);
+    }
+    if ((dataNeeded & _shaderNeeded) != 0 && _curShader != paint.shader) {
+      final Shader? shader = paint.shader;
+      if (shader == null) {
+        _addOp(_CanvasOp.clearShader, 0);
+      } else {
+        _addOp(_CanvasOp.setShader, 0);
+        _addShader(shader);
+      }
+      _curShader = paint.shader;
+    }
+    if ((dataNeeded & _colorFilterNeeded) != 0 && _curColorFilter != paint.colorFilter) {
+      final _ColorFilter? filter = paint.colorFilter?._toNativeColorFilter();
+      if (filter == null) {
+        _addOp(_CanvasOp.clearColorFilter, 0);
+      } else {
+        _addOp(_CanvasOp.setColorFilter, 0);
+        _addColorFilter(filter);
+      }
+      _curColorFilter = paint.colorFilter;
+    }
+    if ((dataNeeded & _imageFilterNeeded) != 0 && _curImageFilter != paint.imageFilter) {
+      final _ImageFilter? filter = paint.imageFilter?._toNativeImageFilter();
+      if (filter == null) {
+        _addOp(_CanvasOp.clearImageFilter, 0);
+      } else {
+        _addOp(_CanvasOp.setImageFilter, 0);
+        _addImageFilter(filter);
+      }
+      _curImageFilter = paint.imageFilter;
+    }
+    if ((dataNeeded & _maskFilterNeeded) != 0 && _curMaskFilter != paint.maskFilter) {
+      final MaskFilter? filter = paint.maskFilter;
+      if (filter == null) {
+        _addOp(_CanvasOp.clearMaskFilter, 0);
+      } else {
+        _addOp(_maskFilterOps[filter._style.index], 1);
+        _addScalar(filter._sigma);
+      }
+      _curMaskFilter = paint.maskFilter;
+    }
+  }
+
+  @override
+  void drawPaint(Paint paint) {
+    _updatePaintData(paint, _paintMask);
+    _addOp(_CanvasOp.drawPaint, 0);
+  }
+
+  @override
+  void drawColor(Color color, BlendMode blendMode) {
+    _addOp(_CanvasOp.drawColor, 2);
+    _addInt(color.value);
+    _addInt(blendMode.index);
+  }
+
+  @override
+  void drawLine(Offset p1, Offset p2, Paint paint) {
+    _updatePaintData(paint, _strokeMask);
+    _addOp(_CanvasOp.drawLine, 2 * _nOffsetData);
+    _addOffset(p1);
+    _addOffset(p2);
+    _addBounds(Rect.fromPoints(p1, p2), true);
```

Comment:
Can this use `_addLTRBToBounds` and avoid allocating a `Rect`? This applies to many other places in this file.

Reply:
`Rect.fromPoints` sorts the points. If I wanted to call directly to LTRB, I'd have to manually sort them here. I can't use `addPoint` because it doesn't adjust for stroking. One thing that addLTRB does that I don't need is that it will include the "minor diagonal corners" of the "rect". So, perhaps I should just have an "add stroked point" method and save both creating a Rect and also adding "false" corners of my "fake rect".

Reply:
I went with an _addStrokedPointToBounds() method.